src/Standards/Generic/Sniffs/PHP/DeclareStrictTypesSniff.php

```diff
@@ -0,0 +1,211 @@
+<?php
+/**
+ * Checks that strict types are declared in the PHP file.
+ *
+ * @author    Michał Bundyra <contact@webimpress.com>
+ * @copyright 2006-2017 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Standards\Generic\Sniffs\PHP;
+
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Util\Tokens;
+
+class DeclareStrictTypesSniff implements Sniff
+{
+    /**
+     * Comment with one of these tags will be omitted. The strict_types
+     * declaration will be placed the next line below the comment.
+     * Otherwise it will be placed line below PHP opening tag.
+     *
+     * @var array
+     */
+    public $omitCommentWithTags = [
+        '@author',
+        '@copyright',
+        '@license',
+    ];
+
+    /**
+     * How declaration should be formatted.
+     *
+     * @var string
+     */
+    public $format = 'declare(strict_types=1);';
+
+    /**
+     * Number of blank lines before declaration.
+     *
+     * @var integer
+     */
+    public $linesBefore = 0;
+
+    /**
+     * Number of blank lines after declaration.
+     *
+     * @var integer
+     */
+    public $linesAfter = 1;
+
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return [T_OPEN_TAG];
```

Comment:
Do you believe that it would be nice to have a check not to run this if the PHP version is below php 7.

I think that if someone uses this sniff inside a ruleset and has a travis that he checks his code against php 5 and 7 it will be hard for him to include/exclude this sniff for the specific version

Reply:
@gmponos I don't think it's needed, because this sniff should be included in your ruleset only when your project supports PHP 7.0+. Otherwise it has no sense.

Reply:
🤦‍♂️ you are correct.